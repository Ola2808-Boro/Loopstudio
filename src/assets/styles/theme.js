export const theme = {
  colors: {
    white: "#ffffff",
    black: "	#000000",
    red: "#FF0000",
    darkGray: "	#8c8c8c",
    veryDarkGray: "#696969",
    vividMulberry: "#C505EB",
    lavenderIndigo: "#8647F8",
  },
  fontSizes: {
    s: "15px",
    m: "20px",
    l: "25px",
    xl: "30px",
  },
};
