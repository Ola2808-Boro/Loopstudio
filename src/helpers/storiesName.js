export const stories = {
  atoms: "Components/atoms/",
  molecules: "Components/molecules/",
  organisms: "Components/organisms/",
  temapltes: "Components/temapltes/",
};
