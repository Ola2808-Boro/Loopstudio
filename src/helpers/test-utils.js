import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { theme } from "assets/styles/theme";
import GlobalStyle from "assets/styles/GlobalStyle";
import DataProvider from "providers/DataProvider";
const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <DataProvider>
        <GlobalStyle />
        {children}
      </DataProvider>
    </ThemeProvider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
