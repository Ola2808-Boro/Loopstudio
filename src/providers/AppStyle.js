import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "assets/styles/theme";
import GlobalStyle from "assets/styles/GlobalStyle";
import "assets/styles/fonts.css";

const AppStyle = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};

export default AppStyle;
