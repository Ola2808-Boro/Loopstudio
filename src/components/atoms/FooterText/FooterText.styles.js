import styled from "styled-components";
import { device } from "helpers/breakpoints";

export const StyledFooterText = styled.h1`
  color: ${({ theme }) => theme.colors.darkGray};
  font-family: "Josefin Sans", sans-serif;
  font-size: ${({ theme }) => theme.fontSizes.s};
  padding: 20px 0px 10px 0px;
`;
