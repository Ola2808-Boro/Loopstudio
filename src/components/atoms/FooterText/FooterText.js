import React, { useContext } from "react";
import { dataContext } from "providers/DataProvider";
import { StyledFooterText } from "components/atoms/FooterText/FooterText.styles";
const FooterText = () => {
  const { data } = useContext(dataContext);

  return <StyledFooterText>{data.footer.text}</StyledFooterText>;
};

export default FooterText;
