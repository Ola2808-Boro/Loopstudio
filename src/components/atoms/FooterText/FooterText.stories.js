import FooterText from "./FooterText";
import withMock from "storybook-addon-mock";
import { stories } from "helpers/storiesName";
import { withReactContext } from "storybook-react-context";
import { dataContext } from "providers/DataProvider";
export default {
  title: `${stories.atoms}FooterText`,
  component: FooterText,
  decorators: [
    withReactContext({
      Context: dataContext,
      initialState: {
        data: {},
        setData: () => {},
        isClicked: false,
        setIsClicked: () => {},
        handleClick: () => {},
        products: {},
        setProducts: () => {},
        click: false,
        isClickedProductDetails: () => {},
      },
    }),
    withMock,
  ],
};

const Template = (args) => <FooterText {...args} />;

export const Default = Template.bind({});

Default.args = {
  data: {
    footer: {
      text: "text",
    },
  },
};

Default.parameters = {
  mockData: [
    {
      url: "https://jsonplaceholder.typicode.com/todos/1",
      method: "GET",
      status: 200,
      response: {
        data: "Hello storybook-addon-mock!",
      },
    },
  ],
};
