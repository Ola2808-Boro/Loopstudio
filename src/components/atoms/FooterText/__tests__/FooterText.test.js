import { render, screen } from "helpers/test-utils";
import renderer from "react-test-renderer";
import { StyledFooterText } from "../FooterText.styles";
import FooterText from "components/atoms/FooterText/FooterText";
import AppStyle from "providers/AppStyle";
describe("test Footer text", () => {
  it("default style", () => {
    const tree = renderer
      .create(
        <AppStyle>
          <StyledFooterText />
        </AppStyle>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  //
});
