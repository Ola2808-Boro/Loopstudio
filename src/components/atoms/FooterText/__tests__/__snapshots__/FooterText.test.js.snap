// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Footer text default style 1`] = `
<h1
  className="sc-bczRLJ hFHjbF"
/>
`;
